File: DSA/LeetCode/Heap/2530. Maximal Score After Applying K Operations.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var maxKelements = function (nums, k) {

    const q = new MaxPriorityQueue();
    for (let a of nums)
        q.enqueue(a)


    let score = 0
    while (k > 0) {
        let n = q.dequeue()?.element ?? 0

        score += n


        q.enqueue(Math.ceil(n / 3))
        k--
    }



    return score

};

File: DSA/LeetCode/Heap/2530. Maximal Score After Applying K Operations.py
import math
class Solution(object):
    def maxKelements(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        pq = [] 
        for a in nums:
            heappush(pq, -a)

        heapq.heapify(pq)
        score = 0 

        while k > 0:
            n = -1*heappop(pq)             
            score += n            
            heappush(pq, -math.ceil(n / 3.00))
            k -=1
            
        return int(score)